### PR TITLE
[#6402] tests: refactor budget list to add test for uncovered lines

### PR DESCRIPTION
**Pull request by phillimorland**

### CODE REVIEW

File: meinberlin/apps/budgeting/assets/__tests__/EndSessionLink.jest.jsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  // overwrite global.fetch with mock function
+  global.fetch = jest.fn().mockRejectedValue('testing: expected network error')
+
+  render(<EndSessionLink />)
+  // toggle modal
+  const modalBtn = screen.getByRole('button', { name: 'End session' })
+  fireEvent.click(modalBtn)
+
+  // get click 'End Session' btn in modal (2nd instance of btn)
+  const endSessionBtn = await screen.getAllByText('End session', { selector: 'button' })[1]
```
Comment (phillimorland) [inline meinberlin/apps/budgeting/assets/__tests__/EndSessionLink.jest.jsx:29]: @goapunk do you know why`find/query/getAllByRole('button', { name: 'End session' })[1]` wouldn't work and returns undefined when console.logged? it's more a point of annoyance but role is a good option to check if something is there

File: package.json [hunk 2]
```diff
@@ -15,6 +15,7 @@
   "dependencies": {
     "@fortawesome/fontawesome-free": "6.3.0",
     "@maplibre/maplibre-gl-leaflet": "0.0.19",
+    "@testing-library/user-event": "^14.4.3",
```
Comment (fuzzylogic2000) [inline package.json]: We don't use the ^, but renovate will also remove that once it#s merged.

Comment (fuzzylogic2000) [reply package.json]: You remove it again in the second commit. :)

File: meinberlin/apps/budgeting/assets/__tests__/EndSessionLink.jest.jsx (hunk 1 above)
Comment (goapunk) [reply meinberlin/apps/budgeting/assets/__tests__/EndSessionLink.jest.jsx:29]: It works if you also include the  hidden option:
```
screen.getAllByRole('button', {name:'End session', hidden: true})[1]   
```

Comment (goapunk) [reply meinberlin/apps/budgeting/assets/__tests__/EndSessionLink.jest.jsx:29]: the `await` here is unnecessary, it's only needed if you use `findBy*`

Comment (phillimorland) [reply meinberlin/apps/budgeting/assets/__tests__/EndSessionLink.jest.jsx:29]: @goapunk so if it is only finding the hidden button and not rendering the modal and finding the button we should delete this right? my understanding is that a test should mimic user behavior, and a user cannot click a hidden button therefore we shouldn't have the test?

**Review by goapunk — APPROVED** [review]
cool!

**Pull request merged**